query/executor.go

```diff
@@ -68,19 +68,35 @@ func ErrMaxConcurrentQueriesLimitExceeded(n, limit int) error {
 	return fmt.Errorf("max-concurrent-queries limit exceeded(%d, %d)", n, limit)
 }
 
-// Authorizer determines if certain operations are authorized.
-type Authorizer interface {
+// CoarseAuthorizer determines if certain operations are authorized at the database level.
+//
+// It is supported both in OSS and Enterprise
+type CoarseAuthorizer interface {
 	// AuthorizeDatabase indicates whether the given Privilege is authorized on the database with the given name.
 	AuthorizeDatabase(p influxql.Privilege, name string) bool
+}
+
+type openCoarseAuthorizer struct{}
+
+func (a openCoarseAuthorizer) AuthorizeDatabase(influxql.Privilege, string) bool { return true }
 
-	// AuthorizeQuery returns an error if the query cannot be executed
-	AuthorizeQuery(database string, query *influxql.Query) error
+//func (a openCoarseAuthorizer) AuthorizeQuery(database string, query *influxql.Query) error { return nil }
 
+// OpenCoarseAuthorizer is a fully permissive implementation of CoarseAuthorizer
+var OpenCoarseAuthorizer openCoarseAuthorizer
+
+// FineAuthorizer determines if certain operations are authorized at the series level.
```

Comment:
I went with Coarse and Fine because in addition to databases we also authorise generic operations with the coarse authoriser (like, 'are you allowed to create users'). I could be convinced to switch the name though.